### PR TITLE
Add dynamic inventory for oVirt version 4 and RHV version 4

### PR DESCRIPTION
**Pull request by machacekondra**
##### ISSUE TYPE

- Feature Pull Request
##### COMPONENT NAME

contrib/inventory/ovirt4.py
##### ANSIBLE VERSION

```
ansible --version
ansible 2.1.1.0
```
##### SUMMARY

This patch adds new dynamic inventory script for oVirt version 4.

```
 $ contrib/inventory/ovirt4.py --host rhel7 --pretty
{
  "ansible_ssh_host": "10.34.60.149", 
  "cluster": "mycluster", 
  "description": null, 
  "devices": {
    "eth0": [
      "10.34.60.149", 
      "fe80::21a:4aff:fe16:151", 
      "2620:52:0:223c:21a:4aff:fe16:151"
    ]
  }, 
  "fqdn": "mydomain.local", 
  "host": "pm", 
  "id": "dba0125e-f53d-4b87-8b42-7ce8bfa6a0e4", 
  "name": "rhel7", 
  "os_type": "rhel_7x64", 
  "statistics": {
    "cpu.current.guest": 1.0, 
    "cpu.current.hypervisor": 0.0, 
    "cpu.current.total": 1.0, 
    "memory.buffered": 2789376.0, 
    "memory.cached": 253116416.0, 
    "memory.free": 862937088.0, 
    "memory.installed": 1073741824.0, 
    "memory.used": 204010946.0, 
    "migration.progress": 0.0
  }, 
  "status": "up", 
  "tags": [], 
  "template": "rhel7"
}

```

### CODE REVIEW

**Comment by machacekondra** [comment]
ready_for_review

**Comment by machacekondra** [comment]
@ryansb @mwperina

**Comment by machacekondra** [comment]
ready_for_review

File: contrib/inventory/ovirt4.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    )
+    parser.add_argument(
+        '--list',
+        action='store_true',
+        default=True,
+        help='Get data of all virtual machines (default: True).',
+    )
+    parser.add_argument(
+        '--host',
+        help='Get data of specific virtual machine.',
```
Comment (mwperina) [inline contrib/inventory/ovirt4.py]: I'd rephrase comment to: 'Get data of virtual machines running on specified host'

File: contrib/inventory/ovirt4.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    """
+    # Get the path of the configuration file, by default use
+    # 'ovirt.ini' file in script directory:
+    default_path = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        'ovirt.ini',
+    )
+    config_path = os.environ.get('OVIRT_INI_PATH', default_path)
+
+    # Create parser and add ovirt section if it don't exists:
```
Comment (mwperina) [inline contrib/inventory/ovirt4.py]: Please fix: '... if it doesn't exist:'

File: contrib/inventory/ovirt4.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    config = ConfigParser.SafeConfigParser()
+    if not config.has_section('ovirt'):
+        config.add_section('ovirt')
+    config.read(config_path)
+
+    # Create connection with varibles defined in ini file:
```
Comment (mwperina) [inline contrib/inventory/ovirt4.py]: I'd rephrase: 'Create a connection with options defined in ini file:'

File: contrib/inventory/ovirt4.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        'devices': dict(
+            (device.name, [ip.address for ip in device.ips]) for device in devices
+        ),
+        'ansible_host': devices[0].ips[0].address if len(devices) > 0 else None,
+    }
+
+
+def get_data(connection, vm_name=None):
+    """
+    Obtain data of `vm_name` if passed, otherwise obtain data of all vms.
```
Comment (mwperina) [inline contrib/inventory/ovirt4.py]: I'd rephrase: 'Obtain data of `vm_name` if specified, ...'

**Review by mwperina — CHANGES_REQUESTED** [review]

File: contrib/inventory/ovirt4.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    """
+    # Get the path of the configuration file, by default use
+    # 'ovirt.ini' file in script directory:
+    default_path = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        'ovirt.ini',
+    )
+    config_path = os.environ.get('OVIRT_INI_PATH', default_path)
+
+    # Create parser and add ovirt section if it doesn't exists:
```
Comment (mwperina) [inline contrib/inventory/ovirt4.py]: 'exists' -> 'exist'

**Review by mwperina — CHANGES_REQUESTED** [review]

**Review by mwperina — COMMENTED** [review]
Looks good to me, thanks! :+1:

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks! :+1:
